src/core/text/Text.js

```diff
@@ -130,48 +131,11 @@ export default class Text extends Sprite
 
         this._font = Text.getFontStyle(style);
 
-        this.context.font = this._font;
-
-        // word wrap
-        // preserve original text
-        const outputText = style.wordWrap ? this.wordWrap(this._text) : this._text;
-
-        // split text into lines
-        const lines = outputText.split(/(?:\r\n|\r|\n)/);
-
-        // calculate text width
-        const lineWidths = new Array(lines.length);
-        let maxLineWidth = 0;
-        const fontProperties = Text.calculateFontProperties(this._font);
-
-        for (let i = 0; i < lines.length; i++)
-        {
-            const lineWidth = this.context.measureText(lines[i]).width + ((lines[i].length - 1) * style.letterSpacing);
-
-            lineWidths[i] = lineWidth;
-            maxLineWidth = Math.max(maxLineWidth, lineWidth);
-        }
-
-        let width = maxLineWidth + style.strokeThickness;
-
-        if (style.dropShadow)
-        {
-            width += style.dropShadowDistance;
-        }
+        const measured = TextMetrics.measure(this._text, this._style, this._style.wordWrap, this.canvas);
```

Comment:
You _may_ want to consider destructuring the metrics object here since you're already doing it with `width` and `height`.

```js
const { width, height, lines, lineHeight, lineWidths, maxLineWidth, fontProperties } = TextMetrics.measure(
    this._text, this._style, this._style.wordWrap, this.canvas
);
```